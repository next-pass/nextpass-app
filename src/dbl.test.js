import {
  _add,
  _update,
  _delete
} from './dbl';

let state = undefined;

it('1- _add', () => {
  state = _add([], {_id: '1', name: 'foo'});
  expect(state).toMatchSnapshot();
});

it('2- _add', () => {
  state = _add(state, {_id: '2', name: 'bar'});
  expect(state).toMatchSnapshot();
});

it('3- _add', () => {
  state = _add(state, {_id: '3', name: 'baz'});
  expect(state).toMatchSnapshot();
});


it('4- update', () => {
  state = _update(state, '2', {name: 'lorem ipsum', notes: 'foo bar baz'});
  expect(state).toMatchSnapshot();
});


it('5- delete', () => {
  state = _delete(state, '1');
  expect(state).toMatchSnapshot();
});
