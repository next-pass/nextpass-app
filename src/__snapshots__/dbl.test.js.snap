// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- _add 1`] = `
Array [
  Object {
    "_id": "1",
    "name": "foo",
  },
]
`;

exports[`2- _add 1`] = `
Array [
  Object {
    "_id": "1",
    "name": "foo",
  },
  Object {
    "_id": "2",
    "name": "bar",
  },
]
`;

exports[`3- _add 1`] = `
Array [
  Object {
    "_id": "1",
    "name": "foo",
  },
  Object {
    "_id": "2",
    "name": "bar",
  },
  Object {
    "_id": "3",
    "name": "baz",
  },
]
`;

exports[`4- update 1`] = `
Array [
  Object {
    "_id": "2",
    "name": "lorem ipsum",
    "notes": "foo bar baz",
  },
  Object {
    "_id": "1",
    "name": "foo",
  },
  Object {
    "_id": "3",
    "name": "baz",
  },
]
`;

exports[`5- delete 1`] = `
Array [
  Object {
    "_id": "2",
    "name": "lorem ipsum",
    "notes": "foo bar baz",
  },
  Object {
    "_id": "3",
    "name": "baz",
  },
]
`;
